Submitted By: Alexander E. Patrakov (semzx at newmail dot ru)
Date: 2003-08-13
Initial Package Version: 3.1.3
Origin: Alexander E. Patrakov
Description: Add support for the old "DOS" IBM-866 charset to KDE.
Applies to Russians only. Not submitted to the KDE team yet.

diff -ur kdelibs-3.1.3/kdecore/kcharsets.cpp kdelibs-3.1.3.tweaked/kdecore/kcharsets.cpp
--- kdelibs-3.1.3/kdecore/kcharsets.cpp	2003-07-14 01:27:41.000000000 +0600
+++ kdelibs-3.1.3.tweaked/kdecore/kcharsets.cpp	2003-08-02 17:46:04.000000000 +0600
@@ -37,7 +37,7 @@
 
 #include <assert.h>
 
-#define CHARSETS_COUNT 33
+#define CHARSETS_COUNT 34
 
 static const char * const language_names[] = {
 	I18N_NOOP( "Other" ),
@@ -107,6 +107,8 @@
     "tscii",                 "tscii", 0,
     "pt 154",                "pt 154","cp 1251","koi8-u","koi8-r","iso8859-5", 0,
     "winsami2",              "winsami2", "cp1252", "unicode", 0,
+    "ibm866",               "ibm866","cp 1251","koi8-u","koi8-r","iso8859-5", 0,
+
     0 }; // extra 0 for end
 
 // 0 other
@@ -171,6 +173,7 @@
     { "ucs2", 15 },
     { "iso-10646-ucs-2", 15 },
     { "winsami2", 16},
+    { "ibm866", 6},
     { 0, 0 } };
 
 // defines some different names for codecs that are built into Qt.
@@ -212,6 +215,8 @@
     { "shift-jis", "sjis" },
     { "x-sjis", "sjis" },
     { "iso-2022-jp", "jis7" },
+    { "cp 866", "ibm866" },
+    { "cp-866", "ibm866" },
     { "windows1250", "cp 1250" },
     { "windows1251", "cp 1251" },
     { "windows1252", "cp 1252" },
